backend/src/src-search/src/main/java/org/eclipse/sw360/search/db/AbstractDatabaseSearchHandler.java

```diff
@@ -88,32 +120,90 @@ public List<SearchResult> search(String text, User user) {
+                typeField.add("name");
+            }
+            if (typeMask.contains("license") || typeMask.contains("user") || typeMask.contains("vendor")) {
+                typeField.add("fullname");
+            }
+            if (typeMask.contains("obligations")) {
+                typeField.add("title");
+            }
+            query = "( " + Joiner.on(" OR ").join(FluentIterable.from(typeMask).transform(addType)) + " ) AND " + "( "
+                    + Joiner.on(" OR ").join(FluentIterable.from(typeField).transform(addField)) + " ) ";
+        }
+        return getFilteredSearchResults(query, user);
     }
 
     /**
      * Search the database for a given string and types
      */
     public List<SearchResult> search(String text, final List<String> typeMask, User user) {
-
-        if (typeMask == null || typeMask.isEmpty()) {
-            return search(text, user);
+        String query = text;
+        if (typeMask != null && !typeMask.isEmpty() && typeMask.get(typeMask.size() - 1).equals("document")) {
+            if (typeMask.size() == 1) {
+                return search(query, user);
+            }
+            typeMask.remove(typeMask.size() - 1);
+            final Function<String, String> addType = input -> "type:" + input;
+            query = "( " + Joiner.on(" OR ").join(FluentIterable.from(typeMask).transform(addType)) + " ) AND "
+                    + prepareWildcardQuery(text);
+            return getSearchResults(query, user);
         }
 
         final Function<String, String> addType = input -> "type:" + input;
-        String query = "( " + Joiner.on(" OR ").join(FluentIterable.from(typeMask).transform(addType)) + " ) AND " + prepareWildcardQuery(text);
-        return getSearchResults(query, user);
+        final Function<String, String> addField = input -> input + ": (" + prepareWildcardQuery(text) + " )";
+        List<String> typeField = new ArrayList<String>();
+        if (typeMask == null || typeMask.isEmpty()) {
+            typeField.add("name");
+            typeField.add("fullname");
+            typeField.add("title");
+            query = "( " + Joiner.on(" OR ").join(FluentIterable.from(typeField).transform(addField)) + " ) ";
+        } else {
+            if (typeMask.contains("project") || typeMask.contains("component") || typeMask.contains("release")) {
+                typeField.add("name");
+            }
+            if (typeMask.contains("license") || typeMask.contains("user") || typeMask.contains("vendor")) {
+                typeField.add("fullname");
+            }
+            if (typeMask.contains("obligations")) {
+                typeField.add("title");
+            }
+            query = "( " + Joiner.on(" OR ").join(FluentIterable.from(typeMask).transform(addType)) + " ) AND " + "( "
+                    + Joiner.on(" OR ").join(FluentIterable.from(typeField).transform(addField)) + " ) ";
+        }
+        return getFilteredSearchResults(query, user);
```

Comment:
this piece of code from line `175` to `195` is exact same as code from line `135` to `155` in method `searchWithoutWildcard`.
Please create a new method and reuse the code.